src/main/java/Duke.java

```diff
@@ -1,10 +1,216 @@
+        System.out.println("Please find your previous list items below: ");
+        int taskIndex=read("/Users/nikki/Documents/GitHub/ip/src/main/java/output.txt");
+        String userInput = in.nextLine();
+
+        while(true) {
+            try {
+                if (userInput.equalsIgnoreCase("bye")) {
+                    System.out.println("Bye. Hope to see you again soon!");
+                    break;
+                }
+
+                int slashIndex;
+                if (userInput.equalsIgnoreCase("list")) {
+                    if (!userInput.equals("list")) {
+                        throw new commandException();
+                    }
+
+                    System.out.println("Here are the tasks in your list: ");
+
+                    for(int index = 0; index < listOfTasks.size(); ++index) {
+                        int j = index + 1;
+                        System.out.println(j + "." + ((Task)listOfTasks.get(index)).getType() +
+                                ((Task)listOfTasks.get(index)).getMark() + " " +
+                                ((Task)listOfTasks.get(index)).toString());
+                    }
+                } else {
+                    PrintStream var10000;
+                    String var10001;
+                    if (userInput.length() > 4 && userInput.contains("delete")) {
+                        slashIndex = Integer.parseInt(userInput.substring(7)) - 1;
+                        var10000 = System.out;
+                        var10001 = ((Task)listOfTasks.get(slashIndex)).getType();
+                        var10000.println("Noted. I've removed this task:\n   " + var10001 +
+                                ((Task)listOfTasks.get(slashIndex)).getMark() + " " + listOfTasks.get(slashIndex));
+                        listOfTasks.remove(slashIndex);
+                        System.out.println("Now you have " + listOfTasks.size() + " tasks in the list.");
+                        save();
+                    } else if (userInput.length() > 4 && userInput.contains("done")) {
+                        slashIndex = Integer.parseInt(userInput.substring(5)) - 1;
+                        ((Task)listOfTasks.get(slashIndex)).Done();
+                        var10000 = System.out;
+                        var10001 = ((Task)listOfTasks.get(slashIndex)).getType();
+                        var10000.println("Nice! I've marked this task as done:\n   " + var10001 + "[✓] " + listOfTasks.get(slashIndex));
+                        save();
+                    } else {
+                        if (userInput.length() > 4 && userInput.contains("todo")) {
+                            ToDos item = new ToDos(userInput.substring(5));
+                            listOfTasks.add(item);
+                            save();
+                        } else if (userInput.length() > 8 && userInput.contains("deadline")) {
+                            slashIndex = userInput.indexOf("by");
+                            int a;
```

Comment:
Maybe you should avoid using the name like "a" for variable. How about "slashIndexOffset"?